labs/firepower-restapi-101/2.md

```diff
@@ -1,67 +1,56 @@
+
+The FMC REST API can perform functions globally, in the default domain, or in specific domains. To select the domain, you must have an account with permissions to access the REST API in one or more domains configured on your FMC. Select the desired domain from the **Domains** menu.
 
 ## Obtaining sample code
-You can use the Export Operation button in API Explorer to save the displayed method example as a Python or Perl script file to your local host. This script can be used to make that REST call, and can be integrated into your application. This is meant primarily as an educational and bootstrapping tool.
 
-Python: The Python scripts require you to install Python, available from https://www.python.org/. Once you’ve installed Python, you can run your script with python script.py username password.
-Perl: Using the Perl scripts requires some additional set-up—you need five components:
+The API Explorer can generate sample REST call code in Python and Perl. This code can be adapted to your specific inmplementation.
+
+1. In the **API INFO** menu, select a resource.
+2. Locate the desired function and select **DELETE**, **PUT**, **POST**, or **GET**.
+3. Select and input parameters for the function.
+4. In the panel on the right, locate the **Export operation in** drop-down menu and select **Python script** or **Perl script**.
+5. The sample code will pop up in a new window. Cut and paste this code to use in your REST API client.
+
+### Code prerequisites
+
+The Python scripts require you to install Python, available from https://www.python.org/. Once you’ve installed Python, you can run your script with `python *script*.py *username* *password*`.
+
+Perl scripts require. additional set-up. You will need to install the following five components:
+
 * Perl package, found at http://www.perl.org/
 * Bundle::CPAN, found at http://search.cpan.org/~andk/Bundle-CPAN-1.861/CPAN.pm
 * REST::Client, found at http://search.cpan.org/~mcrawfor/REST-Client-88/lib/REST/Client.pm
 * MIME::Base64, found at http://perldoc.perl.org/MIME/Base64.html
 * JSON, found at http://search.cpan.org/~makamaka/JSON-2.90/lib/JSON.pm
 
-The API Explorer can create sample code in Python and Perl to aid in creating a REST API client.
-
-Here are steps to create sample code
-* Select the resource where you want to obtain sample code from the API INFO menu.
-* Locate the desired function and select DELETE, PUT, POST, or GET.
-* Select and input all other parameters for the desired function. This will vary depending on the function.
-* On the right, drop down the menu for "Export operation in" and select "Python script" or "Perl script".
-* The sample code will pop up in a new window. Cut and paste this code to use in your REST API client.
-
-![Figure: FMC REST API Explorer sample code screen shot ](/posts/files/firepower-restapi-101/assets/images/fmcexpcode.png)
+![Figure: FMC REST API Explorer sample code screen shot ](assets/images/fmcexpcode.png)
 
```

Comment:
I'm testing out the Python code that I've exported from the Sandbox, and it's not working, but I'll see what users have to do to get it to work. Seems useful to have a real walkthrough using our Sandbox if possible.

Reply:
Still trying to make some of it work myself. I've added some language to the effect that the user should review the code and modify it to match their application/environment.